### DISCUSSION TITLE
Entailment rules Reasoner

### DISCUSSION THREAD

**Discussion by prerak01** [RDFLib/rdflib#1885] [Q&A] [answered]
Is there any support for reasoning which would create implied axioms from the existing RDF axioms? It is not found in the documentation

**Comment by ghost** [comment, marked as answer]
> Is there any support for reasoning which would create implied axioms from the existing RDF axioms? It is not found in the documentation.

The documentation isn't _that_ lacking, so no, no direct support in the core RDFLib library. Instead you'll need to see if your requirements are satisfied by [owl-rl](https://github.com/RDFLib/OWL-RL) ([available from PyPi](https://pypi.org/project/owlrl/) and also in the [list of RDFLib repositories](https://github.com/orgs/RDFLib/repositories?type=all)).

Variously described as: 

> “OWL-RL and RDFS based RDF Closure inferencing for Python - A simple implementation of the OWL2 RL Profile, as well as a basic RDFS inference, on top of RDFLib. Based mechanical forward chaining”

> “A simple implementation of the OWL2 RL Profile on top of RDFLib: it expands the graph with all possible triples that OWL RL defines.”

> “simple forward chaining rules are used to extend (recursively) the incoming graph with all triples that the rule sets permit (ie, the “deductive closure” of the graph is computed)”

**Comment by prerak01** [comment]
OKay
